latch_config/config.py

```diff
@@ -51,17 +70,92 @@ class LoggingMode(str, Enum):
 T = TypeVar("T")
 
 
-def read_config(x: type[T], env_prefix: str = "") -> T:
+class _IsDataclass(Protocol):
+    __dataclass_fields__: ClassVar[dict[str, Field]]
+
+
+DC = TypeVar("DC", bound=_IsDataclass)
+
+
+class NoValue: ...
+
+
+class NestyError(Exception):
```

Comment:
u could also just steal `DataValidationError` from `latch-data-validation`
tho it is a bit thick